File: client/scalaz/src/main/scala/org/openapitools/client/api/Keyword.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Keyword._

case class Keyword (
  /* Name of the keyword */
  keywordName: Option[String],
/* Key of the keyword (short code) */
  keywordKey: Option[String],
/* Internal keyword ID */
  keywordId: Option[Integer],
/* Date when this keyword was added */
  dateAdded: Option[String])

object Keyword {
  import DateTimeCodecs._

  implicit val KeywordCodecJson: CodecJson[Keyword] = CodecJson.derive[Keyword]
  implicit val KeywordDecoder: EntityDecoder[Keyword] = jsonOf[Keyword]
  implicit val KeywordEncoder: EntityEncoder[Keyword] = jsonEncoderOf[Keyword]
}

File: client/javascript-closure-angular/API/Client/Keyword.js
goog.provide('API.Client.Keyword');

/**
 * @record
 */
API.Client.Keyword = function() {}

/**
 * Name of the keyword
 * @type {!string}
 * @export
 */
API.Client.Keyword.prototype.keywordName;

/**
 * Key of the keyword (short code)
 * @type {!string}
 * @export
 */
API.Client.Keyword.prototype.keywordKey;

/**
 * Internal keyword ID
 * @type {!number}
 * @export
 */
API.Client.Keyword.prototype.keywordId;

/**
 * Date when this keyword was added
 * @type {!string}
 * @export
 */
API.Client.Keyword.prototype.dateAdded;

